lib/core/src/packStruct.cpp

```diff
@@ -1,494 +1,708 @@
+  
+    int unpackDoubleToOutPtr(const void *&inPtr,
+                             void *&outPtr,
+                             int numElement,
+                             const char *name,
+                             irodsProt_t irodsProt);
+  
+    int unpackNatDoubleToOutPtr(const void *&inPtr, void *&outPtr, int numElement);
+  
+    int unpackChildStruct(const void *&inPtr,
+                          packedOutput_t &unpackedOutput,
+                          const packItem_t &myPackedItem,
+                          const packInstruct_t *myPackTable,
+                          int numElement,
+                          irodsProt_t irodsProt,
+                          const char *packInstructInp,
+                          const std::optional<irods::version>& peer_version);
+  
+    int unpackPointerItem(packItem_t &myPackedItem,
+                          const void *&inPtr,
+                          packedOutput_t &unpackedOutput,
+                          const packInstruct_t *myPackTable,
+                          irodsProt_t irodsProt,
+                          const std::optional<irods::version>& peer_version);
+  
+    void* addPointerToPackedOut(packedOutput_t &packedOutput, int len, void *pointer);
+  
+    int unpackNatStringToOutPtr(const void *&inPtr, void *&outPtr, int maxStrLen);
+  
+    int unpackXmlStringToOutPtr(const void*& inPtr,
+                                void*& outPtr,
+                                int maxStrLen,
+                                const char* name,
+                                const std::optional<irods::version>& peer_version);
+  
+    int iparseDependent(packItem_t &myPackedItem);
+  
+    int resolveStrInItem(packItem_t &myPackedItem);
+  
+    int packNullString(packedOutput_t &packedOutput);
+  
+    int getNumStrAndStrLen(const packItem_t &myPackedItem, int &numStr, int &maxStrLen);
+  
+    int getAllocLenForStr(const packItem_t &myPackedItem, const void *inPtr, int numStr, int maxStrLen);
+  
+    int packXmlTag(const char *name, packedOutput_t &packedOutput, int endFlag);
+  
+    int parseXmlValue(const void *&inPtr, const char *name, int &endTagLen);
+  
+    int parseXmlTag(const void *inPtr, const char *name, int flag, int &skipLen);
+
+    int packNopackPointer(void *inPtr,
+                          packedOutput_t &packedOutput,
+                          int len,
+                          const char *name,
+                          irodsProt_t irodsProt);
+
+    auto to_version(const std::string& _peer_version) -> std::optional<irods::version>
+    {
+        if (!_peer_version.empty()) {
+            // Expects _peer_version to be something like "rods4.3.0".
+            static const std::regex pattern{R"_(\s*rods(\d+)\.(\d+)\.(\d+)\s*)_"};
+
+            if (std::smatch m; std::regex_match(_peer_version, m, pattern)) {
+                // The first match is the whole string. If there is a match, then we
+                // expect a match size of four.
```

Comment:
Because 3 digits (major, minor, patch) plus first match?

Reply:
yes, but with 3 numbers.

Reply:
use a better comment

Reply:
is the new comment clearer?

Reply:
it is